R/DataManagement.R

```diff
@@ -1,9 +1,155 @@
+                          convertDates = convertDates,
+                          dateFormat = dateFormat,
+                          bestGuess = bestGuess,
+                          overnight = overnight,
+                          deleteErrors = deleteErrors,
+                          ...)
+    
+    new_base = adjust_overlapping_stays(base = new_base,
+                                        patientID = "pID",#ID
+                                        hospitalID = "hID",#FINESS
+                                        admDate = "Adate",
+                                        disDate = "Ddate",
+                                        maxIteration = 25,
+                                        ...)
+    return(new_base)
+}
+
+#' Check database format
+#'
+#' Function that performs various generic checks to ensure that the database has the correct format
+#'
+#' @param base (data.table)
+#'     A patient discharge database, in the form of a data.table. The data.table should have at least the following columns:
+#'         pID: patientID (character)
+#'         hID: hospitalID (character)
+#'         Adate: admission date (POSIXct, but character can be converted to POSIXct)
+#'         Ddate: discharge date (POSIXct, but character can be converted to POSIXct)
+#' @param deleteMissing (character) How to handle records that contain a missing value in at least one of the four mandatory variables:
+#' "no" (default): do not delete. Stops the function with an error message.
+#' "record": deletes just the incorrect record.
+#' "patient": deletes all records of each patient with one or more incorrect records.
+#' 
+#' @return Returns either an error message, or the database (modified if need be).
+#' @export
+#' 
+checkFormat <- function(base,
+                        deleteMissing = "no")
+{
+    if (!"data.frame" %in% class(base)) {
+        stop("The database must be either a data.frame or a data.table object")
+    } else if (!"data.table" %in% class(base)) {
+        setDT(base)
+        message("Converting database to a data.table object")
+    }
+    
+    # Check missing columns
+    cols = c("pID", "hID", "Adate", "Ddate")
+    missingCols = setdiff(cols, colnames(base))
+    if (length(missingCols)) {
+        stop(paste0("The following column(s) is/are missing: ",
+                    paste0(missingCols, collapse = ", "),
+                    ". Please check that your database contains at least the columns mentionned in the documentation, and that they are in the right format. Names are case sensitive."))
+    }
+
+    # Check format of "pID" and "hID" columns
+    cls = sapply(c("pID", "hID"), function(x) typeof(base[[x]]))
+    wrong = names(cls[cls != "character"])
+    if (length(wrong)) {
+        stop(paste0("The following column(s) is/are not of type 'character': ",
+                    paste0(wrong, collapse = ", "),
+                    "."))
+    }
+
+    # Check for missing values    
```

Comment:
Again, for me it makes more sense to do that here, and to do it for all 4 variables and not just the dates